### PR TITLE
Issue1181 Fix Migration 0012_committee_user_relation

### PR DESCRIPTION
**Pull request by r-peschke**

### CODE REVIEW

File: migrations/migrations/0012_committee_user_relation.py [hunk 1]
```diff
@@ -88,31 +160,23 @@ def handle_user_create_event(
...
+            )
+            for committee_id in cml_committee_ids
         ]
 
     def handle_user_delete_event(
-        self, user_id: int, event: DeleteEvent
+        self, event: DeleteEvent, user: Dict[str, Any], user_id: int
     ) -> Optional[List[BaseEvent]]:
-        user = self.new_accessor.get_model_ignore_deleted(f"user/{user_id}")[0]
+        # user = self.new_accessor.get_model_ignore_deleted(f"user/{user_id}")[0]
```
Comment (reiterl) [inline migrations/migrations/0012_committee_user_relation.py:179]: Mmh, don't you want to delete this line?

File: migrations/migrations/0012_committee_user_relation.py [hunk 2]
```diff
@@ -194,10 +250,12 @@ def handle_user_update_event(
...
         )
         new_cml_committee_ids = list(
-            map(int, event.data.get("committee_$_management_level", []))
+            map(
+                int, event.data.get(f"committee_${cml_permission}_management_level", [])
+            )
         )
         new_committee_ids.update(new_cml_committee_ids)
-        user = self.new_accessor.get_model_ignore_deleted(f"user/{user_id}")[0]
+        # user = self.new_accessor.get_model_ignore_deleted(f"user/{user_id}")[0]
```
Comment (reiterl) [inline migrations/migrations/0012_committee_user_relation.py:258]: see above

File: migrations/migrations/0012_committee_user_relation.py [hunk 3]
```diff
@@ -248,15 +306,8 @@ def handle_user_update_event(
...
-            event.data.pop(f"committee_${committee_id}_management_level")
-        event.data["committee_$_management_level"] = [cml_permission]
-        event.data[
-            f"committee_${cml_permission}_management_level"
-        ] = new_cml_committee_ids
-        events = [
-            event,
+        # create event for committee
+        return [
             *[
```
Comment (reiterl) [inline migrations/migrations/0012_committee_user_relation.py:311]: This is not needed here.

**Review by reiterl — COMMENTED** [review]
Some minor changes. I miss the `DeleteFieldsEvent`. Did you think about it?

File: migrations/migrations/0012_committee_user_relation.py (hunk 1 above)
Comment (r-peschke) [reply migrations/migrations/0012_committee_user_relation.py:179]: done

File: migrations/migrations/0012_committee_user_relation.py (hunk 2 above)
Comment (r-peschke) [reply migrations/migrations/0012_committee_user_relation.py:258]: done

File: migrations/migrations/0012_committee_user_relation.py (hunk 3 above)
Comment (r-peschke) [reply migrations/migrations/0012_committee_user_relation.py:311]: done

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]
I'm still getting this error:

```
Migrations:
Finalize migrations.
Position 1 from MI 12 to MI 13 ...
Position 2 from MI 12 to MI 13 ...
Traceback (most recent call last):
  File "migrations/migrate.py", line 142, in <module>
    sys.exit(main())
  File "migrations/migrate.py", line 132, in main
    handler.execute_command(args.command)
  File "migrations/migrate.py", line 63, in execute_command
    self.handler.finalize()
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/migration_handler.py", line 160, in finalize
    if not self.run_migrations():
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/migration_handler.py", line 90, in run_migrations
    return self.migrater.migrate(
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/migrater.py", line 110, in migrate
    self.run_actual_migrations()
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/migrater.py", line 168, in run_actual_migrations
    self.migrate_position(position, last_position_value)
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/migrater.py", line 221, in migrate_position
    new_events = migration.migrate(
  File "/usr/local/lib/python3.8/site-packages/datastore/migrations/core/base_migration.py", line 71, in migrate
    translated_events = self.migrate_event(event)
  File "/app/migrations/migrations/0012_committee_user_relation.py", line 83, in migrate_event
    event.data.pop(f"committee_${committee_id}_management_level")
KeyError: 'committee_$1_management_level'
```

**Review by reiterl — APPROVED** [review]

**Pull request merged**